apps/src/blockly/addons/blockSvgLimit.ts

```diff
@@ -0,0 +1,144 @@
+import {BlockSvg} from 'blockly';
+
+const BUBBLE_SIZE = 18;
+const HALF_BUBBLE_SIZE = BUBBLE_SIZE / 2;
+/**
+ * Represents a bubble on a Blockly block, displaying the count of blocks remaining
+ * based on a limit initially stated in the toolbox XML.
+ */
+export default class BlockSvgLimit {
+  protected element_: BlockSvg;
+  protected count: number;
+  protected className: string;
+  protected limitGroup_: SVGElement | undefined;
```

Comment:
Why do some of these have suffix underscores, and some don't? (I should probably know this.) I was thinking internal vs. external use, but className doesn't appear to be used outside of the `blockSvgLimit` file?

Reply:
Also, I'm curious about the use of private vs. protected here (I'm not that familiar with the usage of these in JS).

Reply:
A lot of these ended up being `protected` because I had initially copied some boiler plate from `svgFrame` which has two derived classes for blocks and workspaces.